### PR TITLE
add parameter to control where saved json files are saved

### PR DESCRIPTION
**Pull request by mjkkirschner**

### CODE REVIEW

File: test/DynamoCoreTests/SerializationTests.cs [hunk 1]
```diff
@@ -221,14 +221,14 @@ private void CompareWorkspaces(WorkspaceComparisonData a, WorkspaceComparisonDat
...
+            DoWorkspaceOpenAndCompare(customNodeTestPath,"json", ConvertCurrentWorkspaceToJsonAndSave, CompareWorkspaces);
         }
 
         [Test]
         public void AllTypesSerialize()
         {
             var customNodeTestPath = Path.Combine(TestDirectory, @"core\serialization\serialization.dyn");
-            DoWorkspaceOpenAndCompare(customNodeTestPath, ConvertCurrentWorkspaceToJsonAndSave, CompareWorkspaces);
+            DoWorkspaceOpenAndCompare(customNodeTestPath, "json", ConvertCurrentWorkspaceToJsonAndSave, CompareWorkspaces);
         }
```
Comment (ramramps) [inline test/DynamoCoreTests/SerializationTests.cs]: What is this PR about?

**Comment by mjkkirschner** [comment]
Previously both the serialization tests and the ones with non guid ids were overwriting each other when saving files to the temp directory this pr saves them to different folders so we get all the files output. Greg wanted to see the latest output Json files and I noticed this potentially confusing behavior.

**Comment by mjkkirschner** [comment]
@QilongTang @ramramps any objections to bringing this in? It fixes a few bugs and makes the serialization tests produce slightly more coherent output for the non-guid id tests.

**Comment by QilongTang** [comment]
It contains node renaing migration for `DSCore.List.ContainsItem`. Is it also intended?

**Comment by mjkkirschner** [comment]
@QilongTang yep, thats a migration that wasn't added previously.

**Comment by QilongTang** [comment]
@mjkkirschner LGTM then

**Pull request merged**